docs/Migrations.rst

```diff
@@ -0,0 +1,260 @@
+
+However this behaviour is not always good. For instance, neo4j doesn't allow schema and data changes in the same transaction.
+
+To disable this, you can use the ``disable_transactions!`` helper in your migration definition:
+
+.. code-block:: ruby
+
+  class SomeMigration < Neo4j::Migrations::Base
+    disable_transactions!
+
+    ...
+  end
+
+Neo4j::Migrations::Helpers
+------------------
+
+#execute
+~~~~~~
+
+Executes a pure neo4j cypher query, interpolating parameters.
+
+.. code-block:: ruby
+
+  execute('MATCH (n) WHERE n.name = {node_name} RETURN n', node_name: 'John')
+
+.. code-block:: ruby
+
+  execute('MATCH (n)-[r:`friend`]->() WHERE n.age = 7 DELETE r')
+
+
+#query
+---------------
+
+An alias for ``Neo4j::Session.query``. You can use it as root for the query builder:
+
+.. code-block:: ruby
+
+  query.match(:n).where(name: 'John').delete(:n)
+
+
+#remove_property
+---------------
+
+Removes a property given a label.
+
+:Ruby:
+  .. code-block:: ruby
+
+    remove_property(:User, :money)
+
+:Cypher:
+  .. code-block:: cypher
+
+    MATCH (n:`User`) REMOVE n.money
+
+#rename_property
+------
+
+Renames a property given a label.
+
+:Ruby:
+  .. code-block:: ruby
+
+    rename_property(:User, :name, :first_name)
+
+:Cypher:
+  .. code-block:: cypher
+
+    MATCH (n:`User`) SET n.first_name = n.name REMOVE n.name
+
+#drop_nodes
+------
+
+Removes all nodes with a certain label
+
+:Ruby:
+  .. code-block:: ruby
+
+    drop_nodes(:User)
+
+:Cypher:
+  .. code-block:: cypher
+
+    MATCH (n:`User`) OPTIONAL MATCH (n)-[r]-() DELETE r,n
+
+#add_label
+------
+
+Adds a label to nodes, given their current label
+
+:Ruby:
+  .. code-block:: ruby
+
+    add_label(:User, :Person)
+
+:Cypher:
+  .. code-block:: cypher
+
+    MATCH (n:`User`) SET n:`Person`
+
+#add_labels
+------
+
+Adds labels to nodes, given their current label
+
+:Ruby:
+  .. code-block:: ruby
+
+    add_label(:User, [:Person, :Boy])
+
+:Cypher:
+  .. code-block:: cypher
+
+    MATCH (n:`User`) SET n:`Person`:`Boy`
+
+#remove_label
+------
+
+Removes a label from nodes, given a label
+
+:Ruby:
+  .. code-block:: ruby
+
+    remove_label(:User, :Person)
+
+:Cypher:
+  .. code-block:: cypher
+
+    MATCH (n:`User`) REMOVE n:`Person`
+
+#remove_labels
+------
+
+Removes labels from nodes, given a label
+
+:Ruby:
+  .. code-block:: ruby
+
+    remove_label(:User, [:Person, :Boy])
+
+:Cypher:
+  .. code-block:: cypher
+
+    MATCH (n:`User`) REMOVE n:`Person`:`Boy`
+
+#rename_label
+------
+
+Renames a label
+
+:Ruby:
+  .. code-block:: ruby
+
+    rename_label(:User, :Person)
+
+:Cypher:
+  .. code-block:: cypher
+
+    MATCH (n:`User`) SET n:`Person` REMOVE n:`Person`
+
+#drop_constraint
+------
+
+Drops an unique constraint on a given label attribute.
+
+**Warning** it would fail unless you define ``disable_transactions!`` in your migration file.
+
+.. code-block:: ruby
+
+  drop_constraint(:User, :name)
```

Comment:
Currently... yes. And maybe it's not a great idea, though.
Check:
https://github.com/neo4jrb/neo4j/pull/1197/files#diff-ab636d4950487b612dfbe9fde69e387bR48

It's better to leave the neo4j error pass through